Firestore/core/src/firebase/firestore/util/comparison.h

```diff
@@ -353,6 +353,35 @@ class Comparable {
   }
 };
 
+/**
```

Comment:
Unlike `CompareContainer`, which is a slam dunk, this optimization is less certain. Let me know if you think it's justified by the numbers I measured.

Reply:
I think the principal here is reasonable, but rather than delegate to a new, nonstandard method we might define a `Comparable` variant that just adds inequality operators without defining `operator==` at all. Then types can define it for themselves. Maybe `InequalityComparable`? WDYT?

Reply:
Done. I'm not sure how much I like `InequalityComparable` as a name, but I cannot think of a better suggestion either.

Reply:
Yeah I'm in the same boat--I couldn't think of anything better either.